src/main/java/org/folio/rest/impl/InitAPIs.java

```diff
@@ -35,15 +51,43 @@ public void init(Vertx vertx, Context context, Handler<AsyncResult<Boolean>> res
         DatabindCodec.mapper().setConfig(deserializationConfig);
         DatabindCodec.prettyMapper().setConfig(deserializationConfig);
         SpringContextUtil.init(vertx, context, ApplicationConfig.class);
-        handler.complete();
+        SpringContextUtil.autowireDependencies(this, context);
+        deployConsumersVerticles(vertx).onSuccess(car -> {
+            handler.handle(Future.succeededFuture());
+            LOGGER.info("Consumer Verticles were successfully started");
+          })
+          .onFailure(th -> {
+            handler.handle(Future.failedFuture(th));
+            LOGGER.error("Consumer Verticles were not started", th);
+          });
       },
       result -> {
         if (result.succeeded()) {
           resultHandler.handle(Future.succeededFuture(true));
         } else {
-          logger.error("Failure to init API", result.cause());
+          LOGGER.error("Failure to init API", result.cause());
           resultHandler.handle(Future.failedFuture(result.cause()));
         }
       });
   }
+
+  private Future<?> deployConsumersVerticles(Vertx vertx) {
+    AbstractApplicationContext springContext = vertx.getOrCreateContext().get(SPRING_CONTEXT_KEY);
```

Comment:
I believe i have seen TODO's in mod-source-record-manager code to update this approach with Spring. Do you know what the enhancement would be OR how to fix the TODOs?

Reply:
I need some investigation into this. I hope I can work with this in the scope of future tasks.